src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenReadonlyStructTests.cs

```diff
@@ -1548,23 +1548,30 @@ void verify(CSharpCompilation comp)
                 var s1 = comp.GetMember<NamedTypeSymbol>("S1");
                 Assert.False(s1.GetMethod("M1").IsDeclaredReadOnly);
                 Assert.False(s1.GetMethod("M1").IsEffectivelyReadOnly);
+                Assert.Equal(RefKind.Ref, s1.GetMethod("M1").ThisParameter.RefKind);
 
                 Assert.True(s1.GetMethod("M2").IsDeclaredReadOnly);
                 Assert.True(s1.GetMethod("M2").IsEffectivelyReadOnly);
+                Assert.Equal(RefKind.RefReadOnly, s1.GetMethod("M2").ThisParameter.RefKind);
 
                 Assert.True(s1.GetProperty("P1").GetMethod.IsDeclaredReadOnly);
                 Assert.True(s1.GetProperty("P1").GetMethod.IsEffectivelyReadOnly);
+                Assert.Equal(RefKind.RefReadOnly, s1.GetProperty("P1").GetMethod.ThisParameter.RefKind);
                 Assert.False(s1.GetProperty("P1").SetMethod.IsDeclaredReadOnly);
                 Assert.False(s1.GetProperty("P1").SetMethod.IsEffectivelyReadOnly);
+                Assert.Equal(RefKind.Ref, s1.GetProperty("P1").SetMethod.ThisParameter.RefKind);
 
                 Assert.True(s1.GetProperty("P2").GetMethod.IsDeclaredReadOnly);
                 Assert.True(s1.GetProperty("P2").GetMethod.IsEffectivelyReadOnly);
+                Assert.Equal(RefKind.RefReadOnly, s1.GetProperty("P2").GetMethod.ThisParameter.RefKind);
 
                 Assert.True(s1.GetProperty("P3").GetMethod.IsDeclaredReadOnly);
                 Assert.True(s1.GetProperty("P3").GetMethod.IsEffectivelyReadOnly);
+                Assert.Equal(RefKind.RefReadOnly, s1.GetProperty("P3").GetMethod.ThisParameter.RefKind);
```

Comment:
Consider extracting a helper method for readability. (The last parameter might be `RefKind?` to allow for static methods.)
```
verifyReadOnly(s1.GetProperty("P3").GetMethod, true, true, RefKind.RefReadOnly);
```